substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -1872,6 +1882,35 @@ pub mod pallet {
 			ensure_signed(origin)?;
 			Self::do_payout_stakers_by_page(validator_stash, era, page)
 		}
+
+		/// Migrates an account's `RewardDestination::Controller` to
+		/// `RewardDestination::Account(controller)`.
+		///
+		/// Effects will be felt instantly (as soon as this function is completed successfully).
+		///
+		/// This will waive the transaction fee if the `payee` is successfully migrated.
+		#[pallet::call_index(27)]
+		#[pallet::weight(T::WeightInfo::set_payee())]
+		pub fn update_payee(
+			origin: OriginFor<T>,
+			controller: T::AccountId,
+		) -> DispatchResultWithPostInfo {
+			let _ = ensure_signed(origin)?;
+			let ledger = Self::ledger(StakingAccount::Controller(controller.clone()))?;
+
+			if !Payee::<T>::contains_key(&ledger.stash) {
+				return Ok(Pays::Yes.into())
+			}
```

Comment:
The `get` below would return the default `Staked` variant if `Payee` has no entry, which would be different from `DeprecatedController`, so I don't think we need this check.